### PR TITLE
Add meta info for clients to store custom data in

### PR DESCRIPTION
**Pull request by Syer10**
Also change some objects to classes so that they are not taking up memory for the lifetime of the application.
The custom data is key value pairs, they will be available through the referenced manga or chapter object as normal json key values
For example if you added reader mode(with the key juiReaderMode) such as webtoon to a manga object, this is what will show up when you request that manga from the api again
```json
{
    "id": 10,
    "title": "Isekai manga",
    "meta": {
        "juiReaderMode": "webtoon"
    }
}
```

### CODE REVIEW

**Comment by Syer10** [comment]
Gradle wrapper check doesnt seem to be my fault

File: server/src/main/kotlin/suwayomi/server/database/migration/M0007_MangaAndChapterMeta.kt [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+import suwayomi.tachidesk.model.table.MangaTable
+
+/*
+ * Copyright (C) Contributors to the Suwayomi project
+ * 
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+class M0007_MangaAndChapterMeta : Migration() {
```
Comment (AriaMoradi) [inline server/src/main/kotlin/suwayomi/server/database/migration/M0007_MangaAndChapterMeta.kt]: M0007 is taken, update your clone

File: server/src/main/kotlin/suwayomi/tachidesk/TachideskAPI.kt [hunk 2]
```diff
@@ -207,6 +221,19 @@ object TachideskAPI {
             ctx.status(200)
         }
 
+        // used to modify a chapter's meta paramaters
```
Comment (AriaMoradi) [inline server/src/main/kotlin/suwayomi/tachidesk/TachideskAPI.kt:224]: chapter -> manga

**Review by AriaMoradi — CHANGES_REQUESTED** [review]

File: server/src/main/kotlin/suwayomi/server/database/migration/M0007_MangaAndChapterMeta.kt (hunk 1 above)
Comment (Syer10) [reply server/src/main/kotlin/suwayomi/server/database/migration/M0007_MangaAndChapterMeta.kt]: Done

File: server/src/main/kotlin/suwayomi/tachidesk/TachideskAPI.kt (hunk 2 above)
Comment (Syer10) [reply server/src/main/kotlin/suwayomi/tachidesk/TachideskAPI.kt:224]: Done

File: server/src/main/kotlin/suwayomi/server/database/migration/M0007_MangaAndChapterMeta.kt (hunk 1 above)
Comment (Syer10) [reply server/src/main/kotlin/suwayomi/server/database/migration/M0007_MangaAndChapterMeta.kt]: Actually not, once sec

Comment (Syer10) [reply server/src/main/kotlin/suwayomi/server/database/migration/M0007_MangaAndChapterMeta.kt]: Now its done

**Pull request merged**